### PR TITLE
Sending the local properies as string so the server.

### LINKED ISSUE #2099: "show properties;" throws java.lang.UnsupportedOperationException after modifying a property
From the CLI, "show properties;" normally provides the output
```
ksql> show properties;

 Property                                               | Default override | Effective Value
-------------------------------------------------------------------------------------------------------------------------------------------
 ksql.extension.dir                                     |                  | ext
 ksql.functions.substring.legacy.args                   |                  | false
 ksql.output.topic.name.prefix                          |                  |
 ksql.persistent.prefix                                 |                  | query_
 ksql.schema.registry.url                               |                  | http://localhost:8081
 ksql.service.id                                        |                  | default_
 ksql.sink.partitions                                   |                  | 4
 ksql.sink.replicas                                     |                  | 1
 ksql.sink.window.change.log.additional.retention       |                  | 1000000
 ksql.statestore.suffix                                 |                  | _ksql_statestore
 ksql.streams.application.id                            | SERVER           | KSQL_REST_SERVER_DEFAULT_APP_ID
 ksql.streams.auto.offset.reset                         |                  | latest
 ksql.streams.bootstrap.servers                         | SERVER           | localhost:9092
 ksql.streams.cache.max.bytes.buffering                 | SERVER           | 10000000
 ksql.streams.commit.interval.ms                        | SERVER           | 2000
 ksql.streams.default.deserialization.exception.handler | SERVER           | io.confluent.ksql.errors.LogMetricAndContinueExceptionHandler
 ksql.streams.num.stream.threads                        | SERVER           | 4
 ksql.transient.prefix                                  |                  | transient_
 ksql.udf.collect.metrics                               |                  | false
 ksql.udf.enable.security.manager                       |                  | true
 ksql.udfs.enabled                                      |                  | true
-------------------------------------------------------------------------------------------------------------------------------------------
```

but not after I update a property:

```
ksql> set 'ksql.sink.replicas' = '2';
Successfully changed local property 'ksql.sink.replicas' to '2'. Use the UNSET command to revert your change.
ksql> show properties;
java.lang.UnsupportedOperationException
```

### PR DESCRIPTION
**Pull request by hjafarpour**
### Description 
Fixes #2099 
In 5.1 we added more capability to check the correctness of the properties in KSQL CLI. But we changed the way we send the properties in KSQLRequest and instead of sending all properties in String format for value we enforced the correct format depending on the property name. This results in a problem with types like `short` when we send the request to the rest endpoint in JSON format. For instance, the server will read the number as `integer` instead of `short` and we will receive `UnsupportedOperationException`.
This PR goes back to send the property values in string format. We still perform checks for the new properties user passes through `set` command. 

In the long run we need to update the server side properties management to perform the same checks for the properties that are passed via the properties files and also enforce the types when reading from the REST endpoint and read from the command line.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalProperties.java [hunk 1]
```diff
@@ -44,9 +44,20 @@ public LocalProperties(final Map<String, Object> initial) {
...
   public Object set(final String property, final Object value) {
+    return set(property, value, true);
+  }
+
+  /**
+   * Set property.
+   *
+   * @param property the name of the property
+   * @param value the value for the property
+   * @return the previous value for the property, or {@code null}.
```
Comment (apurvam) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalProperties.java]: Need to document the 'useParsed` argument.

Comment (hjafarpour) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/properties/LocalProperties.java]: Added docs for the new argument.

**Comment by hjafarpour** [comment]
@apurvam we don't send `short` or other data types to the rest endpoint with this PR and all value formats are String. As I mentioned in the description this PR only changes the behavior back to the way we did to address the regression issue. For more comprehensive solution we should improve how we enforce the value types in servers side and case the read values into the correct type if the received format is incorrect.

**Comment by big-andy-coates** [comment]
I've pushed a different solution to this issue in PR #2118

**Comment by hjafarpour** [comment]
Closing this in favor of #2118

**Pull request closed without merging**